### PR TITLE
47 adding things to the user entities

### PR DESCRIPTION
**Pull request by czz0120**
1. modified class User into abstract class.
2. Made class User a child class of Account.
3. Modified the constructors of three user classes.

### CODE REVIEW

**Review by Tristal25 — DISMISSED** [review]

File: src/main/java/entities/RegUserFactory.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package entities;
+
+public class RegUserFactory {
+    public User creat(String password, String accountName, Profile profile){
```
Comment (alexyin0429) [inline src/main/java/entities/RegUserFactory.java:4]: Typo: `public User create(String password, String accountName, Profile profile){`

File: src/main/java/entities/Review.java [hunk 2]
```diff
@@ -1,71 +1,40 @@
 package entities;
 
-public class Review implements Savable{
+public class Review {
     private int rating;
     private String comment;
```
Comment (alexyin0429) [inline src/main/java/entities/Review.java:5]: if the comment should not be changed later, then `private final String comment` would be better

File: src/main/java/entities/Review.java [hunk 3]
```diff
@@ -1,71 +1,40 @@
 package entities;
 
-public class Review implements Savable{
+public class Review {
     private int rating;
```
Comment (alexyin0429) [inline src/main/java/entities/Review.java:4]: for rating, same suggestion as below

File: src/main/java/entities/User.java [hunk 4]
```diff
@@ -1,5 +1,66 @@
...
 
-abstract class User {
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+
+abstract class User extends Account{
+
+    private Profile profile;
+    private ArrayList<User> blocked = new ArrayList<>();
```
Comment (alexyin0429) [inline src/main/java/entities/User.java:10]: `private List<User> blocked = new ArrayList<>();` Sorry, I forgot to change this to List<User>. Could you please modify it for me? Thx

**Review by alexyin0429 — CHANGES_REQUESTED** [review]

**Review by alexyin0429 — DISMISSED** [review]
Looks gooooood to me

**Review by ryancys1234 — APPROVED** [review]
Looks good! I notice that UserFactory has been replaced by RegUserFactory, meaning that every user starts out as a regular user (and may turn into VipUser later on). This is a good decision.

File: src/main/java/entities/User.java [hunk 5]
```diff
@@ -1,5 +1,87 @@
...
+        while (LocalDateTime.now().compareTo(expire) < 0){
+            this.restrictedTime = Duration.between(expire, LocalDateTime.now()).toSeconds();
+            if (this.restrictedTime > 60) {
+                System.out.println("More than 1 minutes");
+            }else{
+                System.out.println(this.restrictedTime);
+            }
+        }
+        return true; // will return true if restriction is cleared.
+    }
```
Comment (alexyin0429) [inline src/main/java/entities/User.java:75]: `countDownRestrictionTime` is added on Nov14. Need to be reviewed. This method is for counting down and updating the restriction time on this user.

**Review by alexyin0429 — APPROVED** [review]

**Review by yifeiiZhang — APPROVED** [review]
good job

**Review by clarahqy — APPROVED** [review]
looks good to me

**Review by ryancys1234 — APPROVED** [review]
Will need to confirm with Alex about countDownTimeRestriction in the User class, but overall okay!

**Pull request merged**